Formula/b/build2.rb

```diff
@@ -0,0 +1,149 @@
+             "config.cc.loptions=-L#{pkgconf.lib}",
+             "build2/exe{b}"
+      mv "build2/b", "build2/b-boot"
+    end
+
+    # Local build.
+    #
+    # Note that we don't build hermetically to avoid baking Homebrew-specific
+    # paths into the final dylibs.
+    #
+    # Note also that we currently disable private installation since combined
+    # with #{rpath} it causes `brew linkage` failures. See brew issue #16001
+    # for the reasons. The easiest solution is probably to enable relocatable
+    # installation (config.install.relocatable=true) on our side which will
+    # additionally produce relocatable pkg-config files. But for now let's
+    # not complicate things.
+    #
+    loptions = "-L#{pkgconf.lib}"
+    loptions += " -L#{sqlite.lib}" if OS.linux?
+
+    # Work around invisible/inaccessible /usr/lib/libsqlite3.dylib.
+    # Specifically, on recent versions of Mac OS, /usr/lib/ is no longer
+    # accessible and in order to link to system libraries one has to go
+    # through .tbd files ("text-based stub libraries") found in
+    # .../MacOS*.sdk/usr/lib/.
+    #
+    # Note: drop this workaround once upgraded to build2 0.17.0 which can
+    # handle .tbd files directly (see commits 391fbd38 and a860ce05).
+    #
+    if OS.mac?
+      mkdir "#{buildpath}/build/lib"
+      ln_s "#{MacOS.sdk_path}/usr/lib/libsqlite3.tbd", "#{buildpath}/build/lib/libsqlite3.dylib"
+      loptions += " -L#{buildpath}/build/lib"
+    end
+
+    system "build2/build2/b-boot", "-V",
+           "config.cxx=#{ENV.cxx}",
+           "config.cc.coptions=-O3",
+           "config.cc.loptions=#{loptions}",
+           "config.bin.lib=shared",
+           "config.bin.rpath='#{rpath}'",
+           # "config.bin.rpath='#{lib}/build2'",
+           "config.install.root=#{prefix}",
+           # "config.install.private=build2",
+           "config.import.libsqlite3=",
+           "config.build2.libpkgconf=true",
+           "config.config.persist='config.*'@unused=drop",
+           "configure"
+
+    system "build2/build2/b-boot", "-v",
+           "-j", ENV.make_jobs,
+           "install:", "build2/", "bpkg/", "bdep/"
+
+    ENV.prepend_path "PATH", bin
+
+    # Install build system modules.
+    #
+    modules = buildpath.glob("libbuild2-*/").map(&:basename)
+
+    system "b", "-v",
+           "-j", ENV.make_jobs,
+           "install:", *modules.map { |m| "#{m}/" }
+  end
+
+  test do
+    # Suppress loading of default options files.
+    #
+    ENV["BUILD2_DEF_OPT"] = "0"
+    ENV["BPKG_DEF_OPT"] = "0"
+    ENV["BDEP_DEF_OPT"] = "0"
+
+    system bin/"bdep", "new", "--type=lib,no-version", "--lang=c++", "libhello"
+    (testpath/"libhello/build/root.build").append_lines("using autoconf")
+    cd "libhello" do
+      system bin/"bdep", "init", "--config-create", "@default", "cc"
+      system bin/"b", "test"
+      system bin/"b", "-V", "noop:", "libhello/" # Show configuration report.
```

Comment:
Can we assert some output here?

Reply:
Added for tool versions as well as project info.